Opening files in new and existing split panes

johngeorgewright opened the pull request:
See #304 

![tree-view-split](https://cloud.githubusercontent.com/assets/1423566/5680771/a5ef1c44-9809-11e4-95e1-644539989728.gif)

Open the selected file in a new split by using: `ctrl-(left|right|up|down)`. And you can open a selected file in to an existing split by using `ctrl-[1-9]` where the number is the pane index.

johngeorgewright commented:
S'alright... managed to gets the tests working :smile:

kevinsawicki commented on package.json:
```diff
@@ -14,7 +14,7 @@
     "fs-plus": "^2.3.0",
     "minimatch": "~0.3.0",
-    "pathwatcher": "^2.0.12",
+    "pathwatcher": "~2.0.12",
```
This should probably stay as `^`, was there a reason you needed to change it?

johngeorgewright replied:
The version that was downloaded was causing an incompatible error to be thrown by atom.

See #304

kevinsawicki commented on spec/tree-view-spec.coffee:
```diff
@@ -15,6 +15,9 @@ waitsForFileToOpen = (causeFileToOpen) ->
 describe "TreeView", ->
   [treeView, root, sampleJs, sampleTxt, workspaceElement] = []
 
+  selectEntry = (path) ->
```
Usually when a file has `path = require 'path'` at the top, we try not to use `path` as a variable name since it conflicts, for example, `path.join` would be tricky to use inside this function if needed.

What do you thinking about calling it `pathToSelect` instead?

johngeorgewright replied:
Sure thing.

johngeorgewright commented:
I've changed the key mapping to use the core `pane:split-*` keys.

I.e: Now you'd use `[ctrl|cmd]-k left` to open a new split left. Opening splits in existing panes remains the same, although you'd use the `cmd-[pane]` in mac.

johngeorgewright commented:
How are you feeling about this request? Something you'd merge?

kevinsawicki commented on lib/tree-view.coffee:
```diff
@@ -117,6 +121,10 @@ class TreeView extends View
      'tree-view:toggle-ignored-names': -> toggleConfig 'tree-view.hideIgnoredNames'
 
+    [0..8].forEach (index) =>
+      atom.commands.add @element, "tree-view:open-selected-entry-pane-#{index + 1}", =>
```
What do you think about making this `tree-view:open-selected-entry-in-pane-#{index + 1}` just so it reads a bit more naturally when in the command palette?

johngeorgewright replied:
That's cool with me.

kevinsawicki commented:
> How are you feeling about this request? Something you'd merge?

Yeah, this is really cool, I just left one minor comment, let me know what you think.

johngeorgewright commented:
Done :smile:

kevinsawicki commented:
Awesome, thanks for this :ship:
